hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -395,15 +395,15 @@ If an integrity error is seen, it is registered in {{< regref "FAULT_STATUS.LCMG
 For program operations, the program data and its associated integrity are propagated into the flash controller.
 If an integrity error is seen, an error is registered in {{< regref "FAULT_STATUS.PROG_INTG_ERR" >}}.
 
-#### ECC Related Read Errors
+#### ECC and ICV Related Read Errors
 
-In addition to transmission integrity errors described above, the flash can also emit read errors based on [ECC checks](#flash-ecc).
+In addition to transmission integrity errors described above, the flash can also emit read errors based on [ECC and ICV checks](#flash-ecc-and-icv).
 
-Flash reliability ECC errors (multi-bit errors) and integrity ECC errors (storage errors) are both reflected as in-band errors to the entity that issued the transaction.
+Flash reliability ECC errors (multi-bit errors) and integrity check errors (integrity check errors) are both reflected as in-band errors to the entity that issued the transaction.
 That means if a host direct read, controller initiated read or hardware initiated read encounters one of these errors, the error is directly reflected in the operation status.
 
-Further, reliability / integrity ECC errors are also captured in {{< regref "FAULT_STATUS" >}} and can be used to generate fatal alerts.
-The reason ECC errors are captured not captured in {{< regref "STD_FAULT_STATUS" >}} is because it is assumed 2-bit errors can occur in real usage.
+Further, reliability ECC / integrity check errors are also captured in {{< regref "FAULT_STATUS" >}} and can be used to generate fatal alerts.
+The reason these are captured not captured in {{< regref "STD_FAULT_STATUS" >}} is because it is assumed 2-bit errors can occur in real usage.
```

Comment:
" ... are captured not captured ..." do you mean "... are not captured..."?

Reply:
Also, I don't understand the comment about 2-bit errors happening in real usage, could you explain?

As for ICV, since it is implemented as ECC there is a non-trivial probability of them missing to flag multi-bit errors. ECC may not be the best mechanism for integrity checks: it seems to need some sort of one-way hash instead. Any thoughts?

Reply:
yeah that's right. Typo on my side.

It just means in real usage, flash is expected to have bit errors that are not a result of attacks.
And yes it is also true that the detection ability is not 100%, so just like the memory ECC, this is best effort. But it is understood it can be missed.  Anything that has a better detection profile will be too heavy for our purposes.